### PR TITLE
Add Method to Retrieve Detailed Product Description and Specification

### PR DESCRIPTION
**Pull request by Nishitbaria**
## Description

This pull request introduces a new method, get_product_details, to the Flyrobu class, enabling the retrieval of detailed product descriptions and specifications. The method uses web scraping techniques to fetch and process the necessary information from the product page.

## Resolves: #531
Changes Made:
- Added the `get_product_details` method to the `Flyrobu` class, which takes a `product_name` parameter and returns a JSON object containing detailed product information.
- Implemented web scraping techniques to extract the product description, features, technical details, and price details from the product page.
- Extracted the product image URL and included it in the JSON response.
- Handled various exceptions and error cases to ensure robustness and graceful failure.

Usage:
To use the new method, create an instance of the `Flyrobu` class and call `get_product_details` with the desired `product_name`. The method will return a JSON object with the product's detailed information.

Example Usage:
```python
# Import the Flyrobu class and json module
from flyrobu import Flyrobu
import json

# Create an instance of the Flyrobu class
flyrobu = Flyrobu()

# Prompt the user to enter the product name
product_name = input("Enter the product name: ")

# Retrieve the product details using the new method
product_details = flyrobu.get_product_details(product_name)

# Print the product details in a formatted JSON representation
print(json.dumps(product_details, indent=4, ensure_ascii=False))
```

The `get_product_details` method enhances the functionality of the `Flyrobu` class by providing a convenient way to access detailed product information on the platform.

## Checklist

> Before submitting this pull request, kindly verify that the ensuing checkpoints have been reached.

- [x] Have you adhered to the repository's defined coding convention rules?
- [x] Have you updated the 'documentation.md' file with the method/function documentation?
- [x] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [x] Has the method/class been added to the documentation (md file)?

## Screenshots

image

## Additional Notes/Comments

> Any additional remarks or suggestions concerning this pull request are welcome.

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

### CODE REVIEW

**Comment by Nishitbaria** [comment]
@nikhil25803 ,please review and merge my pul request

File: src/scrape_up/flyrobu/flyrobu.py [hunk 1]
```diff
@@ -157,3 +158,84 @@ def search(self, keyword):
 
         result = self.outputs
         return result
+
+    def get_product_details(self, product_name):
+      try:
```
Comment (nikhil25803) [inline src/scrape_up/flyrobu/flyrobu.py:163]: Add docstring with proper code sample and example.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

**Comment by Nishitbaria** [comment]
@nikhil25803 ,I have fixed Docstring ,please review and merge my pull request

File: src/scrape_up/flyrobu/flyrobu.py (hunk 1 above)
Comment (Nishitbaria) [reply src/scrape_up/flyrobu/flyrobu.py]: DOne

**Review by nikhil25803 — APPROVED** [review]

**Comment by nikhil25803** [comment]
Great work @Nishitbaria !!
Do not forget to
+ Share your contribution on socials(twitter mainly), tag us and Clueless 🫂
+ Give this repository a star ✨
+ Follow Clueless Community on GitHub ✅

**Pull request merged**